src/utils.rs

```diff
@@ -1,3 +1,9 @@
+#[cfg(feature = "alloc")]
+extern crate alloc;
```

Comment:
Could you move this `extern crate` to `lib.rs` (and remove `#[cfg(feature = "alloc")]`)?